src/v/kafka/server/handlers/topics/validators.h

```diff
@@ -301,6 +301,40 @@ struct iceberg_config_validator {
     }
 };
 
+/*
+ * it's an error to set the cloud topic property if cloud topics development
+ * feature hasn't been enabled.
+ */
+struct cloud_topic_config_validator {
+    static constexpr const char* error_message
+      = "Cloud topics property is invalid, or support for this development "
+        "feature is not enabled.";
+    static constexpr error_code ec = error_code::invalid_config;
+
+    static bool is_valid(const creatable_topic& c) {
+        auto it = std::find_if(
+          c.configs.begin(),
+          c.configs.end(),
+          [](const createable_topic_config& cfg) {
+              return cfg.name == topic_property_cloud_topic_enabled;
+          });
+        if (it == c.configs.end()) {
+            return true;
+        }
+        if (!config::shard_local_cfg().development_enable_cloud_topics()) {
+            return false;
+        }
```

Comment:
this needs an extra check for safety

```
auto is_user_topic = model::is_user_topic(
          model::topic_namespace{model::kafka_namespace, c.name});
```

take a look at write caching for inspiration https://github.com/redpanda-data/redpanda/blob/577595adba7dd3ba9ed64bb8812fb4e22a8d3b8b/src/v/kafka/server/handlers/topics/validators.h#L325

Reply:
> this needs an extra check for safety

safety for what? yeh we probably shouldn't enable it on non-user topics, but the important thing is that this isn't used in production settings. is that property at risk without the extra check you're asking for?

Reply:
maybe it doesn't actually need the check here; can you change a non-user topic property through the kafka/server/handlers?